test/jdk/java/awt/event/MouseEvent/DragMouseEventTest.java

```diff
@@ -0,0 +1,438 @@
+            return new Point(start.x, top - offset);
+        } else {
+            return new Point(start.x, bottom + offset);
+        }
+    }
+
+    private static class DragMouseEventFrame extends Frame {
+        TextArea textarea = new TextArea("TextArea", 20, 30);
+        Label label = new Label("Label");
+        Panel panel = new Panel();
+        List list = new List();
+        Choice choice = new Choice();
+        Button button = new Button("Button");
+        TextField textfield = new TextField("TextField");
+        Checkbox checkbox = new Checkbox("CheckBox");
+        Scrollbar scrollbar = new Scrollbar();
+        Panel centerPanel = new Panel();
+
+        public DragMouseEventFrame() {
+            setTitle("DragMouseEventTest");
+
+            add(centerPanel, BorderLayout.CENTER);
+            centerPanel.setLayout(new FlowLayout());
+
+            add(panel, BorderLayout.NORTH);
+            panel.setLayout(new FlowLayout());
+
+            choice.add("choice item 1");
+            choice.add("choice item 2");
+            choice.add("choice item 3");
+            panel.add(choice);
+
+            Adapter adapter = new Adapter();
+            choice.addMouseMotionListener(adapter);
+            choice.addMouseListener(adapter);
+            choice.addItemListener(adapter);
+
+            adapter = new Adapter();
+            panel.add(label);
+            label.addMouseMotionListener(adapter);
+            label.addMouseListener(adapter);
+
+            adapter = new Adapter();
+            panel.add(button);
+            button.addMouseMotionListener(adapter);
+            button.addMouseListener(adapter);
+            button.addActionListener(adapter);
+
+            adapter = new Adapter();
+            panel.add(checkbox);
+            checkbox.addMouseMotionListener(adapter);
+            checkbox.addMouseListener(adapter);
+            checkbox.addItemListener(adapter);
+
+            adapter = new Adapter();
+            panel.add(textfield, BorderLayout.EAST);
+            textfield.addMouseMotionListener(adapter);
+            textfield.addMouseListener(adapter);
+            textfield.addActionListener(adapter);
+
+            adapter = new Adapter();
+            add(textarea, BorderLayout.EAST);
+            textarea.addMouseMotionListener(adapter);
+            textarea.addMouseListener(adapter);
+
+            adapter = new Adapter();
+            list.add("list item 1");
+            list.add("list item 2");
+            add(list, BorderLayout.SOUTH);
+            list.addMouseMotionListener(adapter);
+            list.addMouseListener(adapter);
+            list.addActionListener(adapter);
+            list.addItemListener(adapter);
+
+            adapter = new Adapter();
+            add(scrollbar, BorderLayout.WEST);
+            scrollbar.addMouseMotionListener(adapter);
+            scrollbar.addMouseListener(adapter);
+
+            setSize(500, 400);
+            setLocation(200, 200);
```

Comment:
It might be better to have the testUI at the center of the screen to prevent any unwanted interactions with screen menu bars.

```suggestion
           setLocationRelativeTo(null);
```